core/os/thread.cpp

```diff
@@ -28,9 +28,9 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-// Define PLATFORM_CUSTOM_THREAD_H in platform_config.h
+// Define PLATFORM_THREAD_OVERRIDE in your platform's `platform_config.h` to use a custom Thread implementation defined in `platform/[your_platform]/platform_thread.h`
```

Comment:
As I suggested in the original PR, I'd simply remove the comment in the `.cpp` to avoid having to keep them in sync. Maybe just leave:
```
#ifndef PLATFORM_THREAD_OVERRIDE // See details in thread.h
```